tools/pylib/boutdata/data.py

```diff
@@ -245,6 +298,60 @@ def __init__(self, filename, name="root"):
                                 pass
 
                         section[line[:eqpos].strip()] = value
+                        
+        # define arrays of x, y, z to be used for substitutions
+        gridfile = None
+        try:
+            self.nx = self["mesh"].evaluate_scalar("nx")
+            self.ny = self["mesh"].evaluate_scalar("ny")
+        except KeyError:
+            try:
+                gridfilename = self["mesh"]["file"]
+            except KeyError:
+                gridfilename = self["grid"]
+            gridfile = DataFile(gridfilename)
```

Comment:
Reading the grid file could easily fail, since the path is relative to where BOUT++ is run, and could give unexpected results. Since its only the grid size which is needed, I wonder if it would be more robust to read (collect) these from the output files? The dmp files are not guaranteed to be in the same place as the options file (BOUT++ `-f` flag), but probably more reliable than the grid file location.

Reply:
Agree it's a good idea to try to be more reliable, especially with relative paths. I've pushed an update to try to address this issue as best as I could think how to. It's a bit ugly, so any suggestions welcome :smile: